app/origin_node.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import "github.com/weaveworks/scope/report"
+
+// OriginNode is a node in the originating report topology. It's a process ID
+// or network host. It's used by the /api/topology/{topology}/{nodeID} handler
+// to generate detailed information. One node from a rendered topology may
+// have multiple origin nodes.
+type OriginNode struct {
+	Table report.Table
+}
+
+func getOriginNode(t report.Topology, id string) (OriginNode, bool) {
+	node, ok := t.NodeMetadatas[id]
+	if !ok {
+		return OriginNode{}, false
+	}
+
+	// The node represents different actual things depending on the topology.
+	// So we deduce what it is, based on the metadata.
+	if _, ok := node["pid"]; ok {
+		return originNodeForProcess(node), true
+	}
+
+	// Assume network host. Could strengthen this guess by adding a
+	// special key in the probe spying procedure.
+	return originNodeForNetworkHost(node), true
+}
+
+func originNodeForProcess(node report.NodeMetadata) OriginNode {
+	rows := []report.Row{
+		{Key: "Host", ValueMajor: node["domain"], ValueMinor: ""},
+		{Key: "PID", ValueMajor: node["pid"], ValueMinor: ""},
+		{Key: "Process name", ValueMajor: node["name"], ValueMinor: ""},
+	}
+	if val, ok := node["docker_id"]; ok {
```

Comment:
I think I'd prefer to push the human names into the node map and ultimately have the come from the process mappers.  But as is its slightly easier to follow (and extend) as a loop like this, I think.  I'm thinking we're going to be adding a lot more in here (docker image id and name, for instance).